_posts/2019-03-11-PC-Engines_firmware-v4-9-0-3.md

```diff
@@ -0,0 +1,150 @@
+---
+layout: post
+title:  "PC Engines apu coreboot Open Source Firmware v4.9.0.3"
+date:   2019-03-11
+categories: Firmware
+---
+# PC Engines apu coreboot Open Source Firmware v4.9.0.3
+
+## Key changes
+
+Mainline:
+
+1. **Enabled** CPU **boost** feature in **runtime** configuration. Some
+   platforms seem to have problems with boost thus we decided to make CPU boost
+   runtime configurable.
+2. We have **upgraded** to coreboot **SDK 1.52** in mainline releases.
+3. Configured **pull-ups** on **WLAN_DISABLE#** pins on the mPCIe connectors
+   which could cause issues with certain modems when floating.
+4. **Reproducible builds limiation** has been **eliminated** by setting iPXE
+   and Memtest86+ fixed revisions. More details [here](https://github.com/pcengines/coreboot/issues/267)
+5. Uploaded SHA256 and signatures of all previous firmware releases. SHA256
+   files and signatures have been uploaded to pcengines.github.io.
+6. **Fixed microcode update** feature, which broke due to the upstream changes
+   in coreboot. More details [here](https://github.com/pcengines/apu2-documentation/issues/75#issuecomment-463067220)
+7. **Added** information about **ECC** memory capability in **SMBIOS tables**
+   on 4GB platforms.
```

Comment:
@miczyg1 do we have any info how it can be checked? Maybe simple asciinema would be fine?

Reply:
@pietrushnic sure I can record one.

Reply:
Added

Reply:
@miczyg1 IIUC we looking for `Error Correction Type: Multi-bit ECC`? It should be done explicit in video e.g. by grepping this sentence. I'm not sure you can do anything with that at this point. BTW I'm not sure whole booting procedure recording is nedded.

Reply:
@pietrushnic I didn't see the point of grepping because, because simple occurrence of ECC string obtained from grepping may give false information IMO. I have attached the whole boot procedure to show it really is apu2, not some random platform. I chose reliability over cast length.

Reply:
@miczyg1 I'm not sure what is the point of discussion. I'm watching that not from firmware engineer perspective. My point was to as fast as possible prove that feature is there, showing the whole boot cycle may cause 90% of interested people will not watch till the end because they will be bored. On the other hand, maybe you can speed up this video?

I'm not sure - if looking for ECC string may cause false information, then what proves this feature was implemented correctly? Grepping is looking for the given string.

IMO all that discussion is redundant and wastes resources. If you don't want to modify asciinema according to my suggestions, then fine - just know my opinion that I'm not fully satisfied with the marketing value that it provides.

Reply:
I have changed the cast.